src/state/state.md

```diff
@@ -0,0 +1,53 @@
+We have several different notions of state, with different lifecycles on each.
+
+The Lab has a number of open "documents".  A document is represented by a top-level widget in the dock area (but possibly not all top-level widgets are documents - we could tear off a part of a document to make a top-level widget in the dock area that is a part of a document...).  Examples of documents include any file that is represented, as well as terminals, chat windows, real-time video conferences, etc.
+
+Lab IDE Session state
+=====================
+
+Plugins should be able to read/write slices of the IDE state. This IDE state should be preserved on refresh, so a Lab IDE session id should be stored in the URL and the state periodically synced to the server.
+
+Initial implementation could be a local storage solution, tied to the URL.  When the lab starts up, it randomly appends a session id, and then stores session information in local storage tied to the url.  In general, the lab would ask the server for a session id if one is not provided?
+
+Question: on a refresh, should Lab IDE session state only inform the initial setup of the environment (and then the URL changes to reflect a new lab sesion id), or should we try to keep two lab sessions in sync, so if you change the dock layout in one browser window, the dock layout in the other browser changes automatically? Or should we make it easy for a user to 'clone' a lab session?   
```

Comment:
I was thinking that the lab sessions would be tied to the user, i.e., the authentication system would prevent you from going to my lab session URL and seeing my session.